x-pack/plugins/endpoint/public/applications/endpoint/types.ts

```diff
@@ -81,57 +100,100 @@ export interface PolicyListState {
 export interface PolicyDetailsState {
   /** A single policy item  */
   policyItem?: PolicyData;
-  /** data is being retrieved from server */
-  policyConfig?: PolicyConfig;
+  /** API error if loading data failed */
+  apiError?: ServerApiError;
   isLoading: boolean;
   /** current location of the application */
   location?: Immutable<EndpointAppLocation>;
+  /** A summary of stats for the agents associated with a given Fleet Agent Configuration */
+  agentStatusSummary: GetAgentStatusResponse['results'];
+  /** Status of an update to the policy  */
+  updateStatus?: {
+    success: boolean;
+    error?: ServerApiError;
+  };
 }
 
 /**
- * Policy Details configuration
+ * Endpoint Policy configuration
  */
 export interface PolicyConfig {
-  windows: WindowsPolicyConfig;
-  mac: MacPolicyConfig;
-  linux: LinuxPolicyConfig;
+  windows: {
+    events: {
+      process: boolean;
+      network: boolean;
+    };
+    /** malware mode can be detect, prevent or prevent and notify user */
+    malware: {
+      mode: string;
+    };
+    logging: {
+      stdout: string;
+      file: string;
+    };
+    advanced: PolicyConfigAdvancedOptions;
+  };
+  mac: {
+    events: {
+      process: boolean;
+    };
+    malware: {
+      mode: string;
+    };
+    logging: {
+      stdout: string;
+      file: string;
+    };
+    advanced: PolicyConfigAdvancedOptions;
+  };
+  linux: {
+    events: {
+      process: boolean;
+    };
+    logging: {
+      stdout: string;
+      file: string;
+    };
+    advanced: PolicyConfigAdvancedOptions;
+  };
 }
 
-/**
- * Windows-specific policy configuration
- */
-interface WindowsPolicyConfig {
-  /** malware mode can be detect, prevent or prevent and notify user */
-  malware: {
-    mode: string;
-  };
-  eventing: {
-    process: boolean;
-    network: boolean;
+interface PolicyConfigAdvancedOptions {
+  elasticsearch: {
+    indices: {
+      control: string;
+      event: string;
+      logging: string;
+    };
+    kernel: {
+      connect: boolean;
+      process: boolean;
+    };
   };
 }
 
 /**
- * Mac-specific policy configuration
+ * Windows-specific policy configuration that is supported via the UI
  */
-interface MacPolicyConfig {
-  /** malware mode can be detect, prevent or prevent and notify user */
-  malware: {
-    mode: string;
-  };
-  eventing: {
-    process: boolean;
-    network: boolean;
-  };
-}
+type WindowsPolicyConfig = Pick<PolicyConfig['windows'], 'events' | 'malware'>;
```

Comment:
interesting, this allows us to define the Windows config in a more concise way.  Would this help us in updating state in any way in terms of satisfying the type checker?  For instance, I want to update linux eventing - OS selection could trip up the type checker, especially as we add more protections.

Reply:
The Linux (UI only) policy properties are defined further below on line 188.
Maybe the naming here should be changed to `UI<os_type>Config`.